src/routes/jira/server/jira-server-url-post.ts

```diff
@@ -3,34 +3,88 @@ import axios from "axios";
+			message: "Something went wrong",
+			statusCode: 200
+		}
+	}
+};
+
+export const getGheErrorMessages = (codeOrStatus: number | string | null) => {
+	switch (codeOrStatus) {
+		case "invalidUrl":
+			return gheServerUrlErrors.codeOrStatus["invalidUrl"];
+		case "ENOTFOUND":
+			return gheServerUrlErrors.codeOrStatus["ENOTFOUND"];
+		case 502:
+			return gheServerUrlErrors.codeOrStatus[502];
+		default:
+			return gheServerUrlErrors.codeOrStatus["default"];
+	}
+};
+
 export const JiraServerUrlPost = async (
 	req: Request,
 	res: Response
 ): Promise<void> => {
-	const { gheServerURL, installationId } = req.body;
+	const { gheServerURL } = req.body;
+	const { id: installationId } = res.locals.installation;
 
 	req.log.debug(`Verifying provided GHE server url ${gheServerURL} is a valid URL`);
 	const isGheUrlValid = isValidUrl(gheServerURL);
 
-	if (isGheUrlValid) {
-		try {
-			const gitHubServerApps = await GitHubServerApp.getAllForGitHubBaseUrl(gheServerURL, installationId);
-
-			if (gitHubServerApps?.length) {
-				req.log.debug(`GitHub apps found for url: ${gheServerURL}. Redirecting to Jira list apps page.`);
-				res.status(200).send({ success: true, moduleKey: "github-list-apps-page" });
-			} else {
-				req.log.debug(`No existing GitHub apps found for url: ${gheServerURL}. Making request to provided url.`);
-				await axios.get(gheServerURL);
-				res.status(200).send({ success: true, moduleKey: "github-app-creation-page" });
-			}
-		} catch (err) {
-			req.log.debug(`Something went wrong: ${gheServerURL}`, err);
-			// TODO - adding error mapping (create confluence page with error codes)
-			res.status(200).send({ success: false, error: "Something went wrong!!!!" });
+	if (!isGheUrlValid) {
+		const { errorCode, message, statusCode } = getGheErrorMessages("invalidUrl");
+		res.status(statusCode).send({ success: false, errorCode, message });
+		req.log.error(`The entered URL is not valid. ${gheServerURL} is not a valid url`);
+	}
+
+	try {
+		const gitHubServerApps = await GitHubServerApp.getAllForGitHubBaseUrl(gheServerURL, installationId);
+
+		if (gitHubServerApps?.length) {
+			req.log.debug(`GitHub apps found for url: ${gheServerURL}. Redirecting to Jira list apps page.`);
+			res.status(200).send({ success: true, moduleKey: "github-list-apps-page" });
+			return;
 		}
-	} else {
-		req.log.error(`Provided gheServerURL ${gheServerURL} is not a valid url`);
-		res.status(200).send({ success: false, error: "Provided URL is not valid." });
+		req.log.debug(`No existing GitHub apps found for url: ${gheServerURL}. Making request to provided url.`);
+		await axios.get(gheServerURL);
+		res.status(200).send({ success: true, moduleKey: "github-app-creation-page" });
+	} catch (err) {
+		req.log.error({ err, gheServerURL }, `Something went wrong`);
+		const { errorCode, message, statusCode } = getGheErrorMessages(err.code || err.response.status);
+		res.status(statusCode).send({ success: false, errorCode, message });
```

Comment:
the usual standard for errors is to return an array of errors, each one with a code and message, so in this case we only return one, but should still be an array:
`{success: false, errors:[{code: errorCode, message}]}`

Reply:
How does the frontend define which error to render? All of them?